### DISCUSSION TITLE
疑似测出来一个分类BUG，请开发人员关注一下

### DISCUSSION THREAD

**Discussion by Zhaidd1160** [zai-org/ChatGLM3#720] [Q&A]
我实在用 FastGPT + ChatGLM3想实现问题分类。
使用的是ChatGLM3-6B模型，对比的是FastGPT里面的FastAI-4k。
情况是这么个情况：
      我发现用**FastGPT 自带的模型问题分类正常**，
     但是切换到**自己部署的ChatGLM3-6B模型  分类就出现异常**，
     找了半天问题，最后不得已 用postman 请求 把两个模型的 分类场景 请求结果 拿到了，如下：
    
![114ebd2d2e3feed87dae110cfbb0c54](https://github.com/THUDM/ChatGLM3/assets/156068149/daab9dee-f7f2-4082-8db2-57cb6022ccb0)
 DEBUG打印结果：
![87e86c72af1ad863cd9bd9fa0009521](https://github.com/THUDM/ChatGLM3/assets/156068149/aaf97879-7edb-4f4f-9f74-966f6d94afdd)

从上面的对比图 **我有点怀疑** ：
      event: appStreamResponse   里面的 cqResult   跟 debug打印的message的 content 取值不一样  导致最后的 event: answer
choices  出错
      最后反应到流程上面 就走错分支了：
![716dc363cb04fea4cf91bc3b8d065e3](https://github.com/THUDM/ChatGLM3/assets/156068149/3978f8a8-a474-4d64-924d-8f3cd52a19a9)
有点代码BUG的感觉,希望是我操作出错了
![image](https://github.com/THUDM/ChatGLM3/assets/156068149/8ea4f321-9f3a-4f72-9d38-f845950f46c7)

**Comment by Zhaidd1160** [comment]
请求JSON如下：
`{
  "history": [],
  "prompt": "生病了 想要请假",
  "modules": [
    {
      "moduleId": "userChatInput",
      "name": "用户问题(对话入口)",
      "avatar": "/imgs/module/userChatInput.png",
      "flowType": "questionInput",
      "position": {
        "x": 157.0422028435296,
        "y": 906.0508784603846
      },
      "inputs": [
        {
          "key": "userChatInput",
          "type": "systemInput",
          "valueType": "string",
          "label": "用户问题",
          "showTargetInApp": false,
          "showTargetInPlugin": false,
          "connected": false
        }
      ],
      "outputs": [
        {
          "key": "userChatInput",
          "label": "用户问题",
          "type": "source",
          "valueType": "string",
          "targets": [
            {
              "moduleId": "w4bckk",
              "key": "userChatInput"
            }
          ]
        }
      ]
    },
    {
      "moduleId": "w4bckk",
      "name": "问题分类",
      "avatar": "/imgs/module/cq.png",
      "flowType": "classifyQuestion",
      "showStatus": true,
      "position": {
        "x": 675.7708165468526,
        "y": 484.06286988287684
      },
      "inputs": [
        {
          "key": "switch",
          "type": "target",
          "label": "core.module.input.label.switch",
          "valueType": "any",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": false
        },
        {
          "key": "model",
          "type": "selectCQModel",
          "valueType": "string",
          "label": "分类模型",
          "required": true,
          "showTargetInApp": false,
          "showTargetInPlugin": false,
          "value": "ChatGlm3",
          "connected": false
        },
        {
          "key": "systemPrompt",
          "type": "textarea",
          "valueType": "string",
          "label": "背景知识",
          "description": "你可以添加一些特定内容的介绍，从而更好的识别用户的问题类型。这个内容通常是给模型介绍一个它不知道的内容。",
          "placeholder": "例如: \n1. AIGC（人工智能生成内容）是指使用人工智能技术自动或半自动地生成数字内容，如文本、图像、音乐、视频等。\n2. AIGC技术包括但不限于自然语言处理、计算机视觉、机器学习和深度学习。这些技术可以创建新内容或修改现有内容，以满足特定的创意、教育、娱乐或信息需求。",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "value": "你是一个企业服务助手，请根据我的对话内容，你要对每一行对话内容进行分类并告诉我结果，不要返回其他信息和多于的文字，这些类别是:\n请假\n出差\n其它\n请选择其中一个类别并返回，你只要返回类别的名称，不要返回其他信息。让我们开始吧:",
          "connected": false
        },
        {
          "key": "history",
          "type": "numberInput",
          "label": "core.module.input.label.chat history",
          "required": true,
          "min": 0,
          "max": 30,
          "valueType": "chatHistory",
          "value": 6,
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": false
        },
        {
          "key": "userChatInput",
          "type": "target",
          "label": "core.module.input.label.user question",
          "required": true,
          "valueType": "string",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": true
        },
        {
          "type": "custom",
          "valueType": "any",
          "label": "",
          "showTargetInApp": false,
          "showTargetInPlugin": false,
          "key": "agents",
          "value": [
            {
              "value": "请假",
              "key": "wqre"
            },
            {
              "value": "出差",
              "key": "sdfa"
            },
            {
              "value": "其它",
              "key": "c9g6"
            }
          ],
          "connected": false
        }
      ],
      "outputs": [
        {
          "key": "wqre",
          "label": "",
          "type": "hidden",
          "targets": [
            {
              "moduleId": "lil7p9",
              "key": "switch"
            }
          ]
        },
        {
          "key": "sdfa",
          "label": "",
          "type": "hidden",
          "targets": [
            {
              "moduleId": "va670i",
              "key": "switch"
            }
          ]
        },
        {
          "key": "c9g6",
          "label": "",
          "type": "hidden",
          "targets": [
            {
              "moduleId": "ztqo6m",
              "key": "switch"
            }
          ]
        },
        {
          "key": "agex",
          "label": "",
          "type": "hidden",
          "targets": []
        }
      ]
    },
    {
      "moduleId": "lil7p9",
      "name": "指定回复",
      "avatar": "/imgs/module/reply.png",
      "flowType": "answerNode",
      "position": {
        "x": 1348.8078338238874,
        "y": 437.78743638706646
      },
      "inputs": [
        {
          "key": "switch",
          "type": "target",
          "label": "core.module.input.label.switch",
          "valueType": "any",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": true
        },
        {
          "key": "text",
          "type": "textarea",
          "valueType": "any",
          "label": "回复的内容",
          "description": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "placeholder": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "value": "请假1",
          "connected": false
        }
      ],
      "outputs": [
        {
          "key": "finish",
          "label": "core.module.output.label.running done",
          "description": "core.module.output.description.running done",
          "valueType": "boolean",
          "type": "source",
          "targets": []
        }
      ]
    },
    {
      "moduleId": "ztqo6m",
      "name": "指定回复",
      "avatar": "/imgs/module/reply.png",
      "flowType": "answerNode",
      "position": {
        "x": 1353.896387031517,
        "y": 1130.024438081086
      },
      "inputs": [
        {
          "key": "switch",
          "type": "target",
          "label": "core.module.input.label.switch",
          "valueType": "any",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": true
        },
        {
          "key": "text",
          "type": "textarea",
          "valueType": "any",
          "label": "回复的内容",
          "description": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "placeholder": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "value": "其它1",
          "connected": false
        }
      ],
      "outputs": [
        {
          "key": "finish",
          "label": "core.module.output.label.running done",
          "description": "core.module.output.description.running done",
          "valueType": "boolean",
          "type": "source",
          "targets": []
        }
      ]
    },
    {
      "moduleId": "va670i",
      "name": "指定回复",
      "avatar": "/imgs/module/reply.png",
      "flowType": "answerNode",
      "position": {
        "x": 1353.0155917021366,
        "y": 785.9057107056809
      },
      "inputs": [
        {
          "key": "switch",
          "type": "target",
          "label": "core.module.input.label.switch",
          "valueType": "any",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "connected": true
        },
        {
          "key": "text",
          "type": "textarea",
          "valueType": "any",
          "label": "回复的内容",
          "description": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "placeholder": "可以使用 \\n 来实现连续换行。\n可以通过外部模块输入实现回复，外部模块输入时会覆盖当前填写的内容。\n如传入非字符串类型数据将会自动转成字符串",
          "showTargetInApp": true,
          "showTargetInPlugin": true,
          "value": "出差1",
          "connected": false
        }
      ],
      "outputs": [
        {
          "key": "finish",
          "label": "core.module.output.label.running done",
          "description": "core.module.output.description.running done",
          "valueType": "boolean",
          "type": "source",
          "targets": []
        }
      ]
    }
  ],
  "variables": {},
  "appId": "659d0d68a6f6b5d33b654129",
  "appName": "调试-1",
  "detail": true,
  "stream": true
}`

**Comment by Zhaidd1160** [comment]
失误，进一步定位 是中间环节的问题，误伤

**Reply by JustStudyOne** [reply]
老兄，请问您这个问题解决了吗，我也出现了这个问题。使用的oneapi做中转
